File: Native/Test/Python/mr-array-generator.py
import fabric

client = fabric.createClient()

cv = client.MR.createConstValue("Size", 10)

ago = client.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { output = sqrt(Scalar(index)); }", "foo")
ag = client.MR.createArrayGenerator(cv, ago)

count = ag.getCount()
print("ag.getCount() = " + str(count))
for i in range(0, count):
  print("ag.produce("+str(i)+") = "+str(ag.produce(i)))

client.close()

File: Native/Test/Node/mr-array-generator.js
/*
 *  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
 */

FC = require("Fabric").createClient();

cv = FC.MR.createConstValue("Size", 10);

ago = FC.KLC.createArrayGeneratorOperator("foo.kl", "operator foo(io Scalar output, Size index) { output = sqrt(Scalar(index)); }", "foo");
ag = FC.MR.createArrayGenerator(cv, ago);

var count = ag.getCount();
console.log("ag.getCount() = " + count);
for (var i=0; i<count; ++i)
  console.log("ag.produce("+i+") = "+ag.produce(i));

FC.close();
